client/src/hooks/useGetAllLeagueSeasons.js

```diff
@@ -2,10 +2,15 @@ import { useQuery } from 'react-query'
 import { getAllLeagueSeasons } from 'api'
 import { extractData, getLeagueFromUser } from 'utils'
 import useUserDetails from './useUserDetails'
+import { useRouter } from 'next/router'
 
 const useGetAllLeagueSeasons = (options = {}) => {
+    const router = useRouter()
+    let leagueId = router.query?.leagueId
     const { user } = useUserDetails()
-    const leagueId = getLeagueFromUser(user)?._id
+    if (!leagueId) {
+        leagueId = getLeagueFromUser(user)?._id
```

Comment:
This is something I actually only just realised, but getLeagueFromUser means we access the leagueId of the user. However, that user object has an array of leagues, along with the array of seasons for all of the leagues, making it kind of useless to actually refetch the seasons again, since we already have that information from the user object?

Reply:
Note that the user object is currently populed with
-> teams -> games (meaning the full teams and games documents are there, not just the ids)
-> leagues -> seasons (meaning full league and seasons documents are there, not just ids)

Thus fromt a quick check, this is the only hook that is kind of redundant. `useGetAllSeasonGrades` doesn't have the execpPopulate grades, so that one is fine.